src/twisted/test/test_process.py

```diff
@@ -1199,6 +1199,86 @@ def checkTerminated(ignored):
 
         return ended
 
+    def _testEnvironment(self, spawnKwargs, expectedEnv):
+        p = Accumulator()
+        d = p.endedDeferred = defer.Deferred()
+        reactor.spawnProcess(
+            p,
+            pyExe,
+            [
+                pyExe,
+                b"-c",
+                networkString(
+                    "import os, sys; "
+                    "env = dict(os.environ); "
+                    # LC_CTYPE is set by python, see https://peps.python.org/pep-0538/
+                    'env.pop("LC_CTYPE", None); '
+                    'env.pop("__CF_USER_TEXT_ENCODING", None); '
+                    "sys.stderr.write(str(sorted(env.items())))"
+                ),
+            ],
+            usePTY=self.usePTY,
+            **spawnKwargs,
+        )
+
+        def processEnded(ign):
+            expectedEnv.pop("LC_CTYPE", None)
+            expectedEnv.pop("__CF_USER_TEXT_ENCODING", None)
+            self.assertEqual(
+                bytes(str(sorted(expectedEnv.items())), "utf-8"),
+                p.outF.getvalue() if self.usePTY else p.errF.getvalue(),
+            )
+
+        return d.addCallback(processEnded)
+
+    def test_environmentPosixSpawnpEnvNotSet(self):
+        """
+        Spawned process serializes its environ to stderr which is checked against
+        os.environ of the calling process.
+        """
+        return self._testEnvironment({}, {})
+
+    def test_environmentForkEnvNotSet(self):
+        """
+        Spawned process serializes its environ to stderr which is checked against
+        os.environ of the calling process.
+        """
+        return self._testEnvironment({"path": "/tmp"}, {})
+
+    def test_environmentPosixSpawnpEnvNone(self):
+        """
+        Spawned process serializes its environ to stderr which is checked against
+        os.environ of the calling process.
+        """
+        return self._testEnvironment({"env": None}, os.environ)
+
+    def test_environmentForkEnvNone(self):
+        """
+        Spawned process serializes its environ to stderr which is checked against
```

Comment:
I am not sure why `fork` is expected to be triggered by this test.

I think that a dedicated docstring could help identity what is tested here and answer the question: "Why we need this test?"

Reply:
Fixed.